src/WebJobs.Script.WebHost/Diagnostics/LinuxAppServiceFileLoggerFactory.cs

```diff
@@ -20,7 +20,7 @@ public LinuxAppServiceFileLoggerFactory()
         public virtual LinuxAppServiceFileLogger GetOrCreate(string category)
         {
             return Loggers.GetOrAdd(category,
-                c => new Lazy<LinuxAppServiceFileLogger>(() => new LinuxAppServiceFileLogger(category, _logRootPath, new FileSystem()))).Value;
+                static (c, path) => new Lazy<LinuxAppServiceFileLogger>(() => new LinuxAppServiceFileLogger(c, path, new FileSystem())), _logRootPath).Value;
```

Comment:
The lambda of the lazy could also be marked as static

Reply:
The lazy bits _are_ using context, so no static there - it'd fail :)

Reply:
Damn I missed that in the diff view. Sorry for that